controller/src/main/java/io/pravega/controller/server/SegmentHelper.java

```diff
@@ -148,27 +160,32 @@ public void processingFailure(Exception error) {
 
             @Override
             public void connectionDropped() {
+                log.warn("deleteSegment {}/{}/{} Connection dropped", scope, stream, segmentNumber);
                 result.completeExceptionally(
                         new WireCommandFailedException(type, WireCommandFailedException.Reason.ConnectionDropped));
             }
 
             @Override
             public void wrongHost(WireCommands.WrongHost wrongHost) {
+                log.warn("deleteSegment {}/{}/{} wrong host", scope, stream, segmentNumber);
                 result.completeExceptionally(new WireCommandFailedException(type, WireCommandFailedException.Reason.UnknownHost));
             }
 
             @Override
             public void noSuchSegment(WireCommands.NoSuchSegment noSuchSegment) {
+                log.info("deleteSegment {}/{}/{} NoSuchSegment", scope, stream, segmentNumber);
```

Comment:
Is it never a problem, or is it possible that a segment disappearing is a problem? This feels like a WARN, but I'm not sure.

Reply:
Since this is in delete segment, its rerun will result in NoSuchSegment. Idempotent handling warrants treating this as success case. Hence info log.